packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -31,41 +35,40 @@ const userSignUp = z
 			})
 			.regex(UserValidationRegexRule.EMAIL_DOMAIN_PART_VALID_CHARS, {
 				message: UserValidationMessage.EMAIL_WRONG,
-			})
-			.regex(UserValidationRegexRule.EMAIL_LOCAL_PART_LENGTH, {
-				message: UserValidationMessage.EMAIL_LOCAL_PART_MAX_LENGTH,
-			})
-			.regex(UserValidationRegexRule.EMAIL_DOMAIN_PART_LENGTH, {
-				message: UserValidationMessage.EMAIL_DOMAIN_PART_MAX_LENGTH,
 			}),
 		name: z
 			.string()
 			.trim()
-			.min(UserValidationRule.NAME_MINIMUM_LENGTH, {
-				message: UserValidationMessage.NAME_REQUIRE,
-			})
-			.max(UserValidationRule.NAME_MAX_LENGTH, {
-				message: UserValidationMessage.NAME_MAX_LENGTH,
+			.min(UserValidationRule.NON_EMPTY_STRING_MIN_LENGTH, {
+				message: UserValidationMessage.FIELD_REQUIRED,
 			})
-			.regex(UserValidationRegexRule.NAME_VALID_CHARS, {
+			.regex(UserValidationRegexRule.NAME_VALID_CHARS_MIN_MAX, {
 				message: UserValidationMessage.NAME_INVALID_CHARACTERS,
 			}),
 		password: z
 			.string()
 			.trim()
-			.min(UserValidationRule.PASSWORD_MIN_LENGTH, {
-				message: UserValidationMessage.PASSWORD_MIN_LENGTH,
-			})
-			.max(UserValidationRule.PASSWORD_MAX_LENGTH, {
-				message: UserValidationMessage.PASSWORD_MAX_LENGTH,
+			.min(UserValidationRule.NON_EMPTY_STRING_MIN_LENGTH, {
+				message: UserValidationMessage.FIELD_REQUIRED,
 			})
+			.regex(
+				UserValidationRegexRule.PASSWORD_CONTAINS_LETTER_NUMBER_AND_LENGTH,
+				{
+					message: UserValidationMessage.PASSWORD_REQUIRES_CHARACTER,
+				},
+			)
 			.regex(UserValidationRegexRule.PASSWORD_VALID_CHARS, {
 				message: UserValidationMessage.PASSWORD_INVALID_CHARACTERS,
-			})
-			.regex(UserValidationRegexRule.PASSWORD_CONTAINS_LETTER_AND_NUMBER, {
-				message: UserValidationMessage.PASSWORD_REQUIRES_LETTER_AND_NUMBER,
 			}),
 	})
-	.required();
-
+	.required()
+	.superRefine(({ confirmPassword, password }, contex) => {
+		if (confirmPassword !== password) {
```

Comment:
Confirm password field handles in onSubmit function in sign up form, It shoud not be here. Delete super refine method.

Reply:
But, according to the document requirement's this field should be show in real time validation
![image](https://github.com/user-attachments/assets/5f39e955-de6d-4c72-8a99-a7e1f4477c5c)